workers/unity/Packages/io.improbable.gdk.mobile/Editor/MobileLauncherWindow.cs

```diff
@@ -48,112 +53,181 @@ public void OnGUI()
                 {
-                    iOSLaunchUtils.Build(launchConfig.DevelopmentTeamId);
+                    iOSLaunchUtils.Build(mobileLaunchConfig.DevelopmentTeamId);
                 }
 
                 CommonUIElements.DrawHorizontalLine(8, DarkGrey);
 
-                using (new GUILayout.HorizontalScope())
+                DisplayDevicesAndLaunchButton(DeviceType.iOSSimulator);
+
+                CommonUIElements.DrawHorizontalLine(8, DarkGrey);
+
+                DisplayDevicesAndLaunchButton(DeviceType.iOSDevice);
+            }
+        }
+#endif
+
+        private void DisplayDevicesAndLaunchButton(DeviceType deviceType)
+        {
+            int index;
+            List<DeviceLaunchConfig> devices;
+
+            switch (deviceType)
+            {
+                case DeviceType.AndroidDevice:
+                    (index, devices) = androidDevices;
+                    break;
+                case DeviceType.AndroidEmulator:
+                    (index, devices) = androidEmulators;
+                    break;
+                case DeviceType.iOSDevice:
+                    (index, devices) = iOSDevices;
+                    break;
+                case DeviceType.iOSSimulator:
+                    (index, devices) = iOSSimulators;
+                    break;
+                default:
+                    throw new ArgumentException($"Invalid Device Type: {deviceType}");
+            }
+
+            var prettyDeviceType = DeviceLaunchConfig.GetPrettyDeviceType(deviceType);
+            var names = devices.Select(config => config.deviceName).ToArray();
+            var noDevicesFound = devices.Count == 0;
+
+            // List of devices/emulators with refresh button linked to correct platform
+            using (new GUILayout.HorizontalScope())
+            {
+                using (new EditorGUI.DisabledScope(noDevicesFound))
                 {
-                    using (new EditorGUI.DisabledScope(simulatorNames.Length == 0))
-                    {
-                        simulatorNameIndex =
-                            EditorGUILayout.Popup("Simulator Model", simulatorNameIndex, simulatorNames);
-                    }
-
-                    var buttonIcon = new GUIContent(EditorGUIUtility.IconContent("Refresh"))
-                    {
-                        tooltip = "Refresh your simulator list."
-                    };
-
-                    if (GUILayout.Button(buttonIcon, EditorStyles.miniButton, GUILayout.ExpandWidth(false)))
-                    {
-                        simulatorNames = iOSLaunchUtils.RetrieveAvailableiOSSimulators().Keys.ToArray();
-                        simulatorNameIndex = 0;
-                    }
+                    SetDeviceIndex(deviceType,
+                        noDevicesFound
+                            ? EditorGUILayout.Popup(prettyDeviceType, 0, emptyDeviceNameList)
+                            : EditorGUILayout.Popup(prettyDeviceType, index, names));
```

Comment:
Instead of this, we could not have a popup when there are no devices and display simple text instead.

Not sure if that will play nice with the layout engine though

Reply:
I think I prefer having the popup still there with this text because it then makes it clear why it's disabled